src/streamlink/plugins/bbciplayer.py

```diff
@@ -75,7 +64,67 @@ def can_handle_url(cls, url):
     def _hash_vpid(cls, vpid):
         return sha1(cls.hash + str(vpid).encode("utf8")).hexdigest()
 
+    @classmethod
+    def _extract_nonce(cls, http_result):
+        """
+        Given an HTTP response from the sessino endpoint, extract the nonce, so we can "sign" requests with it.
+        We don't really sign the requests in the traditional sense of a nonce, we just incude them in the auth requests.
+
+        :param http_result: HTTP response from the bbc session endpoint.
+        :type http_result: requests.Response
+        :return: nonce to "sign" url requests with
+        :rtype: string
+        """
+
+        # Extract the redirect URL from the last call
+        last_redirect_url = urlparse(http_result.history[-1].request.url)
+        last_redirect_query = dict(parse_qsl(last_redirect_url.query))
+        # Extract the nonce from the query string in the redirect URL
+        final_url = urlparse(last_redirect_query['goto'])
+        goto_url = dict(parse_qsl(final_url.query))
+        goto_url_query = parse_json(goto_url['state'])
+
+        # Return the nonce we can use for future queries
+        return goto_url_query['nonce']
+
+    def _validate_login(self, result_history, ptrt_url):
+        """
+        Check that the last redirect in the auth flow, matches our provided snapback URL
+
+        :param result_history:
+        :param ptrt_url: The snapback URL to redirect to after successful authentication
+        :type result_history: list[requests.Response]
+        :type ptrt_url: string
+        :return: Whether authentication was successful
+        :rtype: bool
+        """
+        no_history = len(result_history) == 0
+        bad_goto_url = True
+        if not no_history:
+            goto_url = str(result_history[-1].url)
```

Comment:
_i don't have an uk ip, but the redirections should still be the same._
_if it's geo blocking it would be weird_

your script is doing the exact opposite, what the old script did.

your works for http urls, not https
the old works for https, not http

---


don't use the history as url
use the last url
`res.url`

for
```sh
streamlink http://www.bbc.co.uk/iplayer/live/cbeebies worst
# it will be
res.history[-1].url =  http://www.bbc.co.uk/iplayer/live/cbeebies
res.url = https://www.bbc.co.uk/iplayer/live/cbeebies
```
this login will work but it checks the wrong url

---

```sh
streamlink https://www.bbc.co.uk/iplayer/live/cbeebies worst
# it will be
res.history[-1].url =  https://session.bbc.co.uk/session/callback?code=......
res.url = https://www.bbc.co.uk/iplayer/live/cbeebies
```
this will show an login error even if it works.

Reply:
Fair point regarding the history. I did test it and got streams from the bbc in vlc. I'll outline my motivations.

**Why inspect the history of the response?**
When you perform auth against [https://account.bbc.com/signin](https://account.bbc.com/signin), if the credentials are invalid, then there are no HTTP redirects. You are presented with an HTML page prompting by a picture of David Attenborough sitting on a beach telling you to log in. So, the quickest and (imho) least complicated way to check if it worked, is if you were redirected at all, eg `len(result_history) == 0`.

FYI, the auth flow should work outside the UK, as I think it's part of the BBC's attempt to sell videos overseas. So you should be able to use the BBC id with the account url above.

**Why not check the redirect url against the response?**
I think I have an idea, but I'm going to test what you said above, before I shoot my mouth off and say somthing wrong. :-)

Reply:
Ah! I misunderstood what you were saying. I see it now. I've pushed some changes through.